docs/DeveloperGuide.md

```diff
@@ -1,37 +1,300 @@
+    * Pros: Beginner friendly, easily understandable, no need to remember flags.
+    * Cons: Slower, implementation when user is familiar with the application.
+
+### CalculateCommand feature
+
+To enable users to view their finances in a more meaningfully, MindMyMoney does calculations to present financial data
+that is actionable for the users.
+
+![calculate_command_sequence_diagram](images/gif_loading.gif)
+
+### ListCommand feature
+The source code can be found in [`ListCommand.java`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command/ListCommand.java)  
+
+MindMyMoney allow users to view their current list of added expenditures. 
+#### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when listing.  
+
+![list_command_sequence_diagram](images/List_Command_Sequence_Diagram.png)
+Fig 13 - List Command Sequence Diagram
+
+#### ListCommand design considerations
+Aspect: How to make the `ListCommand` easily tested using JUnit testing.
+* Alternative 1 (current choice): Abstract the conversion of `Expenditure` to `String` in a separate` ListCommand.listToString()` method.
+  * Pros: Easily tested using JUnit testing, by checking the String that the `ListCommand.listToString()` method returns.
+  * Cons: Added layer of abstraction that may be deemed redundant.
+
+* Alternative 2: Print directly in the `ListCommand.executeCommand()` method .
+  * Pros: Easily implemented with lesser lines of code as the code is minimalist.
+  * Cons: JUnit testing would require I/O redirection prior to checking the output matches expectations.
+
+
+
+## Appendix Requirements
+  
+  
+### Product scope
+
+**Target user profile**
+- a student beginner who wants to start managing their finances
+- possess an income
+- prefer desktop application
+- is proficient in CLI
+- can type fast
+- prefers typing to mouse interactions
+
+**Value proposition**
+Manage finances containing multiple payment methods faster than a typical mouse/GUI driven app.
+
+### User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
-## Non-Functional Requirements
+### Non Functional Requirements
+
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.   
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands)
+   should be able to accomplish most of the tasks faster using commands than using the mouse.
 
-{Give non-functional requirements}
 
-## Glossary
+### Glossary
 
-* *glossary item* - Definition
+* Mainstream OS: Windows, Linux, Unix, OS-X  
 
 ## Instructions for manual testing
 
```

Comment:
Kindly do add instructions for manual testing of the features so a new developer can understand how to test these features. You can include positive and negative test cases, expected outcomes.